### DISCUSSION TITLE
imu fusion (opti track data with drone gyro data)

### DISCUSSION THREAD

**Discussion by kluedeugene** [USC-ACTLab/crazyswarm#502] [Q&A]
Hello, whoenig.
I got a lot of help from your project. Thank you very much.
I used OptiTrack to hover the last two CrazyFly and succeeded in executing example scripts such as Waypoint Samples.
I want to match the world coordinate axis with the gyro coordinate axis of the drone.

When the drone is powered on, can it bring up the absolute value of the world coordinate system of the motif and align it with the relative yaw axis of the drone itself?
Do you have any advice to help us?
Thank you.

**Comment by whoenig** [comment]
If you use full pose tracking (i.e., multiple markers per CF), the alignment is done automatically. If you use a single marker, the two will be aligned after a few seconds of flight, as long as you move in the x/y plane. All of this is achieved through the on-board EKF.

**Reply by kluedeugene** [reply]
thanks for your answer.

What I want to know is how to modify the initial yaw value of crazyflie by substituting the offset value and using the world coordinate system data from the optitrack.
I want modify the yaw value when turning on the crazyflie using the offset value as much as the difference value.
Using the offset value, etc.,
Can you give me advice like where to get the these data?

I'm sorry that I'm not fluent in English.  thanks.

**Reply by whoenig** [reply]
I don't fully understand what you mean. If you want to change the initial guess of the yaw estimate, you would need to do that in the crazyflie firmware, not in the Crazyswarm.
